src/internet-computer-programming-concepts/basic-memory-persistence/upgrades.md

```diff
@@ -14,15 +14,18 @@ There are two ways to *update* the Wasm module of an actor running in a canister
 ### Reinstall
 When we have an actor Wasm module running in a canister, we could always reinstall that same actor module or a new actor module inside the same canister. Reinstalling always causes the actor to be 'reset'. Whether reinstalling the same actor Wasm module or a new one, the operation is the same as if installing for the first time.
 
-**Reinstalling always wipes out the main memory of the canister and erases the state of the actor including any stable variables.** 
+**Reinstalling always wipes out the main memory of the canister and erases the state of the actor** 
 
 ### Upgrade
 We could also choose to *upgrade* the Wasm module of an actor. Then:
 - [Stable variables](/internet-computer-programming-concepts/basic-memory-persistence/stable-variables.html) are persisted
-- [Pre and post upgrade hooks](/advanced-concepts/system-apis/preupgrade-postupgrade.html) are run before and after the upgrade
 - The [actor interface](/internet-computer-programming-concepts/async-data/candid.html#actor-interfaces) is checked for [backwards compatibility](/internet-computer-programming-concepts/basic-memory-persistence.html#service-upgrades-and-sub-typing)
+- [Pre and post upgrade hooks](/advanced-concepts/system-apis/preupgrade-postupgrade.html) are run before and after the upgrade
+
```

Comment:
I would re-order these steps:

* interface check
* preupgade system method is run (to give the user a chance to save transient (unstable) state to stable variables, if necessary.
* code is upgraded
* stable variables are restored.
* postupgrade system method is run to give the user a change to restore transient (unstable) state from stable variables, if necessary.